BASE/src/Microsoft.ApplicationInsights/Extensibility/Implementation/TelemetryConfigurationFactory.cs

```diff
@@ -57,6 +57,7 @@ public virtual void Initialize(TelemetryConfiguration configuration, TelemetryMo
             try
             {
                 SdkInternalOperationsMonitor.Enter();
+                SelfDiagnosticsModule.EnsureInitialized();
```

Comment:
Would this be a safer place?
https://github.com/microsoft/ApplicationInsights-dotnet/blob/develop/BASE/src/Microsoft.ApplicationInsights/Extensibility/TelemetryConfiguration.cs#L63

Reply:
By "safer", do you mean it's a place it can be ensured to initialize? Or it can be ensured to initialize only when `System.Diagnostics.DiagnosticSource.dll` is available?

I went through the code. Only [`System.Diagnostics.Tracing`](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.tracing?view=net-5.0) namespace and `System.Diagnostics` namespace (for [`System.Diagnostics.Process`](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process?view=net-5.0)) are used. It doesn't need `System.Diagnostics.DiagnosticSource.dll`.

I've moved it here but after the `TryRun` method because 1. it's a static method, 2. it's called in all code paths before `TelemetryConfigurationFactory.Initialize` anyways.

Reply:
Sorry for not being more specific - I meant to do this initialize inside the static ctor, to ensure its done at the very earliest possible stage.
(Has nothing to do with Sys.Diag package, so no need of doing inside TryRun..) - It looks like you already figured it out :)

Reply:
I've updated it.